### PR TITLE
Filter by panel fix

### LINKED ISSUE #1232: HPO term and Gene Panel combined
When ticking the boxes for HPO term and a default gene panel, the results (variants) depends on the order you choice to "tick"

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR should fix #1232

Looking at how the query works I've noticed that when both a gene list (or the HPO panel) and another panel is provided, then the query looks like this:

{'case_id': 'internal_id', 'category': 'snv', 'variant_type': 'clinical', '**$or': [{'hgnc_symbols':** {'$in': ['EPM2A', 'MSX2', 'NDUFV2', 'EDN3', 'EHMT1', 'KATNB1', 'MLYCD', 'BCR', 'CHN1', 'ARSA', 'SRPX2', 'EEF1A2', 'GABRG2', 'PSPH', 'SIX3', 'GNAS', 'PCYT1A', 'ARID1B', 'ASS1', 'NDE1', 'STAMBP', 'KAT6B', 'HPD', 'DIP2B', 'LYST', 'LAMA2', 'LONP1', 'GNS', 'MKRN3', 'SLC20A2', 'SAMHD1', 'ANK3', 'MPC1', 'PQBP1', 'ADNP', 'TCTN1', 'DAG1', 'TRAPPC9', 'MPDU1', 'BUB3', 'MCCC1', 'PGK1', 'SOX5', 'EBP', 'MUT', 'SPG11', 'TBX1', 'NIN']}}, {**'panels'**: {'$in': ['hpo']}}], '**$and': [{'$or': [{'hgnc_symbols'**: {'$in': ['EPM2A', 'MSX2', 'NDUFV2', 'EDN3', 'EHMT1', 'KATNB1', 'MLYCD', 'BCR', 'CHN1', 'ARSA', 'SRPX2', 'EEF1A2', 'GABRG2', 'PSPH', 'SIX3', 'GNAS', 'PCYT1A', 'ARID1B', 'ASS1', 'NDE1', 'STAMBP', 'KAT6B', 'HPD', 'DIP2B', 'LYST', 'LAMA2', 'LONP1', 'GNS', 'MKRN3', 'SLC20A2', 'SAMHD1', 'ANK3', 'MPC1', 'PQBP1', 'ADNP', 'TCTN1', 'DAG1', 'TRAPPC9', 'MPDU1', 'BUB3', 'MCCC1', 'PGK1', 'SOX5', 'EBP', 'MUT', 'SPG11', 'TBX1', 'NIN']}}, {**'panels':** {'$in': ['hpo']}}]}]}

Which is basically a double query for the same things. Since when you have both genes and panels you are supposed to add them to the general query in the end, I'm removing the line that adds them directly when the gene filter is created, so the final query looks like this:

{'case_id': 'internal_id', 'category': 'snv', 'variant_type': 'clinical', '**$and': [{'$or': [{'hgnc_symbols':** {'$in': ['DAG1', 'EPM2A', 'HPD', 'PCYT1A', 'TBX1', 'SAMHD1', 'NDE1', 'TCTN1', 'MSX2', 'DIP2B', 'NDUFV2', 'SLC20A2', 'EBP', 'EEF1A2', 'GNS', 'ANK3', 'STAMBP', 'BCR', 'BUB3', 'PSPH', 'LAMA2', 'PQBP1', 'SIX3', 'NIN', 'KAT6B', 'SRPX2', 'MLYCD', 'ARID1B', 'KATNB1', 'CHN1', 'MUT', 'SOX5', 'MPDU1', 'GNAS', 'MCCC1', 'MPC1', 'PGK1', 'MKRN3', 'LYST', 'ADNP', 'SPG11', 'ASS1', 'EHMT1', 'LONP1', 'EDN3', 'TRAPPC9', 'GABRG2', 'ARSA']}}, **{'panels': {'$in':** ['hpo']}}]}]}

This was a bug, but I'm not entirely sure that it is fixing the above issue..

- [x] reviewed by @dnil / @northwestwitch
- [x] tested by @dnil

### CODE REVIEW

**Comment by dnil** [comment]
Well found; this indeed looks unnecessary! Upon testing, it does not solve the issue. I would guess there is some interaction with the variants view; will try to have a look as well during the meeting!

**Comment by northwestwitch** [comment]
yes, I suspected it. I'll have another look as well!

**Comment by dnil** [comment]
For testing I would suggest:
- pick a case with a small panel - like TestPanel
- add a (broad) HPO term (e.g. HP:0001251)
- open variants page, search with clinical filter
- filter with the small panel, if it wasn't already the default
- note the variants found
- check also HPO in the panel list and filter again
- ensure that variants from the HPO panel are indeed found - if need be check HPO panel separately and note the ones there (e.g. COL18A1)

**Comment by dnil** [comment]
Ok, I think that may have been the issue. Seems to add up the queries now. What do you think @northwestwitch?

**Comment by dnil** [comment]
Suggestion for testing?

**Comment by northwestwitch** [comment]
Sorry I think I canceled the progress of the review (my comments) by mistake I'll fix it tomorrow morning!

File: scout/server/blueprints/variants/views.py [hunk 1]
```diff
@@ -254,10 +264,54 @@ def sv_variants(institute_id, case_name):
...
+        flash("Gene not included in clinical list: {}".format(", ".join(non_clinical_symbols)), 'warning')
+    form.hgnc_symbols.data = hgnc_symbols
+
+
+    # handle HPO gene list separately
     if form.data['gene_panels'] == ['hpo']:
         hpo_symbols = list(set(term_obj['hgnc_symbol'] for term_obj in
                                case_obj['dynamic_gene_list']))
         form.hgnc_symbols.data = hpo_symbols
+    elif 'hpo' in form.data['gene_panels']:
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/views.py]: Same as above, I guess

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
 ### Added
 
 ### Fixed
-
+- Checking HPO query together with other panel did not extend query.
```
Comment (northwestwitch) [inline CHANGELOG.md]: What about: Fixed variants query by panel (hpo panel + gene panel) ?

**Comment by northwestwitch** [comment]
The code is now on stage. How to test it using scout-stage data:

1. Go to case scout-stage.scilifelab.se/cust002/F0014188 and note the pinned variant (rank=21).
1. Open variants view and filter variants using the 'ataxi' panel. The above variant should not be returned.
1. Add a phenotype (HP:0100556 - Hemiatrophy) to the case and generate its gene panel
1. It should contain the pinned gene from point 1.
1. Filter using HPO panel and note that the pinned variant is returned.
1. Filter using BOTH HPO panel and ataxi panel and check that the variant is returned.(try by first clicking on ataxi panel and then hpo panel in the filters and the other way around).
1. Remove ataxi panel and the variant should still be returned by the hpo panel.
1. Remove hpo panel and its genes in the filters and click on the ataxi panel. The variant should not be returned.

File: CHANGELOG.md (hunk 2 above)
Comment (dnil) [reply CHANGELOG.md]: A synthesis:
"- Fixed variants view filter query: gene panel + HPO dropdown checkbox combination could miss genes."

**Comment by dnil** [comment]
Looks good: haven't heard back from @47KW but tested ok locally before minor change and on stage now with current branch. Good to merge in other words. I took the liberty of removing the "approve deploy" - to me that belongs with the Release? I would not deploy an unversioned release. We can revisit that if you want it any other way!

**Comment by northwestwitch** [comment]
It's fine to remove approve deployment I think. We are not deploying now after all! Thanks for your help fixing this bug!

**Comment by dnil** [comment]
I won't reopen the PR for that, but realised that I had a synthesis for a CHANGELOG statement in a comment : 
"Fixed variants view filter query: gene panel + HPO dropdown checkbox combination could miss genes." 
In case you want it for a release.. 😆

**Pull request merged**